### PR TITLE
Environment variable to disable plugins

### PR DESCRIPTION
**Pull request by geospackle**
## Change Summary
Environment variable `PYDANTIC_DISABLE_PLUGINS` can be used to disable all or specified plugins.

## Related issue number
#7709
## Checklist

* [x] The pull request title is a good summary of the changes - it will be used in the changelog
* [x] Unit tests for the changes exist
* [x] Tests pass on CI
* [x] Documentation reflects the changes where applicable
* [x] My PR is ready to review, **please add a comment including the phrase "please review" to assign reviewers**

Selected Reviewer: @dmontagu

### CODE REVIEW

**Comment by geospackle** [comment]
please review

**Comment by geospackle** [comment]
1. Should we add a warning when plugin is disabled?
2. Is it useful to add default value `0` to enable plugins?

File: docs/concepts/plugins.md [hunk 1]
```diff
@@ -116,3 +116,12 @@ class Foo(BaseModel, plugin_settings={'observer': 'all'}):
 ```
 
 On each validation call, the `plugin_settings` will be passed to a callable registered for the events.
+
+## Disabling Plugins
+You can use environment variable `PYDANTIC_DISABLE_PLUGINS` to disable all or specific plugin(s).
```
Comment (Kludex) [inline docs/concepts/plugins.md:121]:
```suggestion

You can use environment variable `PYDANTIC_DISABLE_PLUGINS` to disable all or specific plugin(s).
```

File: docs/concepts/plugins.md [hunk 2]
```diff
@@ -116,3 +116,12 @@ class Foo(BaseModel, plugin_settings={'observer': 'all'}):
...
+
+
+| Environment Variable          | Allowed Values                                        | Description                   |
+|-------------------------------|-------------------------------------------------------|-------------------------------|
+| `PYDANTIC_DISABLE_PLUGINS`    | `__all__`, `1`                                        | Disables all plugins          |
```
Comment (Kludex) [inline docs/concepts/plugins.md:126]: I guess "true" as well?

File: pydantic/plugin/_loader.py [hunk 3]
```diff
@@ -22,10 +23,13 @@ def get_plugins() -> Iterable[PydanticPluginProtocol]:
 
     Inspired by: https://github.com/pytest-dev/pluggy/blob/1.3.0/src/pluggy/_manager.py#L376-L402
     """
+    disabled_plugins = os.getenv('PYDANTIC_DISABLE_PLUGINS')
     global _plugins, _loading_plugins
     if _loading_plugins:
         # this happens when plugins themselves use pydantic, we return no plugins
         return ()
+    elif disabled_plugins == '__all__' or disabled_plugins == '1' or disabled_plugins == 'true':
```
Comment (Kludex) [inline pydantic/plugin/_loader.py:31]:
```suggestion
    elif disabled_plugins in ('__all__', '1', 'true'):
```

File: docs/concepts/plugins.md [hunk 4]
```diff
@@ -116,3 +116,13 @@ class Foo(BaseModel, plugin_settings={'observer': 'all'}):
 ```
 
 On each validation call, the `plugin_settings` will be passed to a callable registered for the events.
+
+## Disabling Plugins
+
+You can use environment variable `PYDANTIC_DISABLE_PLUGINS` to disable all or specific plugin(s).
+
```
Comment (Kludex) [inline docs/concepts/plugins.md:123]:
```suggestion
```

**Review by Kludex — APPROVED** [review]

**Pull request merged**